bin/strata-client/src/rpc_server.rs

```diff
@@ -636,44 +636,61 @@ impl StrataAdminApiServer for AdminServerImpl {
 }
 
 pub struct SequencerServerImpl {
-    inscription_handle: Arc<InscriptionHandle>,
+    envelope_handle: Arc<EnvelopeHandle>,
     broadcast_handle: Arc<L1BroadcastHandle>,
     checkpoint_handle: Arc<CheckpointHandle>,
     params: Arc<Params>,
 }
 
 impl SequencerServerImpl {
     pub fn new(
-        inscription_handle: Arc<InscriptionHandle>,
+        envelope_handle: Arc<EnvelopeHandle>,
         broadcast_handle: Arc<L1BroadcastHandle>,
         params: Arc<Params>,
         checkpoint_handle: Arc<CheckpointHandle>,
     ) -> Self {
         Self {
-            inscription_handle,
+            envelope_handle,
             broadcast_handle,
             params,
             checkpoint_handle,
         }
     }
-}
 
-#[async_trait]
-impl StrataSequencerApiServer for SequencerServerImpl {
-    async fn submit_da_blob(&self, blob: HexBytes) -> RpcResult<()> {
-        let commitment = hash::raw(&blob.0);
-        let blobintent = BlobIntent::new(BlobDest::L1, commitment, blob.0);
+    /// Submit DA payload entries to be placed in commit reveal Envelope
+    /// multiple Envelopes can exist in same transaction
+    async fn submit_blobs(&self, blob_vec: Vec<EnvelopePayload>) -> RpcResult<()> {
+        let blob_commitment = BundledCommitment::from_payload(&blob_vec);
+        let blobintent = BundledPayloadIntent::new(DataBundleDest::L1, blob_commitment, blob_vec);
         // NOTE: It would be nice to return reveal txid from the submit method. But creation of txs
         // is deferred to signer in the writer module
         if let Err(e) = self
-            .inscription_handle
-            .submit_intent_async(blobintent)
+            .envelope_handle
+            .submit_bundled_intent_async(blobintent)
             .await
         {
             return Err(Error::Other(e.to_string()).into());
         }
         Ok(())
     }
+}
+
+#[async_trait]
+impl StrataSequencerApiServer for SequencerServerImpl {
+    async fn submit_da_blobs(&self, blobs: Vec<HexBytes>) -> RpcResult<()> {
+        let blob_vec: Vec<EnvelopePayload> = blobs
+            .into_iter()
+            .map(|blob| EnvelopePayload::new(PayloadTypeTag::DA, blob.into_inner()))
```

Comment:
As we are adding a new RPC endpoint, could we generalize this to allow for other types of da tag items too ?